src/main/java/core/basesyntax/serviceimpl/ProcessServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ProcessService;
+import core.basesyntax.strategy.Strategy;
+import java.util.List;
+
+public class ProcessServiceImpl implements ProcessService {
+    private final Strategy strategy;
+
+    public ProcessServiceImpl(Strategy strategy) {
+        this.strategy = strategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactionList) {
+        for (FruitTransaction transaction : fruitTransactionList) {
+            strategy.get(transaction.getOperation())
+                    .processOperation(transaction);
+        }
+    }
+}
```

Comment:
The class `ProcessServiceImpl` should not directly depend on the `Strategy` interface. This creates a high level of coupling between the service and the strategy implementation. Instead, consider using a `Map` that would be passed to the `ProcessService` implementation, containing all the necessary strategies for different operations. This would adhere to the Dependency Inversion Principle, one of the SOLID principles, and would make the code more flexible and easier to manage.